2017-18 nba season player stats

In [ ]:
#import libraries
import pandas as pd
import numpy as np

In [ ]:
#load data
df=pd.read_csv('/content/2017-18_NBA_salary.csv')

In [ ]:
# view dataframe
df.head()

Player    Salary NBA_Country  NBA_DraftNumber  ...  OBPM DBPM   BPM  VORP
0        Zhou Qi    815615       China               43  ... -10.6  0.5 -10.1  -0.2
1  Zaza Pachulia   3477600     Georgia               42  ...  -0.6  1.3   0.8   0.7
2  Zach Randolph  12307692         USA               19  ...  -0.6 -1.3  -1.9   0.0
3    Zach LaVine   3202217         USA               13  ...  -0.7 -2.0  -2.6  -0.1
4   Zach Collins   3057240         USA               10  ...  -3.7  0.9  -2.9  -0.2

[5 rows x 28 columns]

Dataset shows: Player name, salary, their team/country, their draft number and their season stats

G= games, MP=minutes played, PER= player efficiency rating, TS,= true shooting percentage, 3PA= 3-point field goal attempt, FTr= free-throw percentage, ORB=offensive rebound, DRB=defensive rebound, BLK= blocks, TOV= turnover, USG= usage%((FGA + 0.44 * FTA + TOV) * (Tm MP / 5)) / (MP * (Tm FGA + 0.44 * Tm FTA + Tm TOV)). Usage percentage is an estimate of the percentage of team plays used by a player while he was on the floor.)
OW= offensive win shares, DWS=defensive win shares, WS=win share(player contribution to wins), OW/48= win share per 48 mins, BPM= box plus/minus (player contribution to team when they are on the court), VORP= value over replacement player


In [ ]:
# stats in df
df.shape

(485, 28)

In [ ]:
# info about df, columns, non-null counts and their datatypes
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 485 entries, 0 to 484
Data columns (total 28 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Player           485 non-null    object 
 1   Salary           485 non-null    int64  
 2   NBA_Country      485 non-null    object 
 3   NBA_DraftNumber  485 non-null    int64  
 4   Age              485 non-null    int64  
 5   Tm               485 non-null    object 
 6   G                485 non-null    int64  
 7   MP               485 non-null    int64  
 8   PER              485 non-null    float64
 9   TS%              483 non-null    float64
 10  3PAr             483 non-null    float64
 11  FTr              483 non-null    float64
 12  ORB%             485 non-null    float64
 13  DRB%             485 non-null    float64
 14  TRB%             485 non-null    float64
 15  AST%             485 non-null    float64
 16  STL%             485 non-null    float64
 17  BLK%            

In [ ]:
# statistal values
df.describe()

Salary  NBA_DraftNumber  ...         BPM        VORP
count  4.850000e+02       485.000000  ...  485.000000  485.000000
mean   6.636507e+06        29.451546  ...   -1.760206    0.598763
std    7.392602e+06        21.125760  ...    5.661447    1.245653
min    4.608000e+04         1.000000  ...  -49.200000   -1.300000
25%    1.471382e+06        11.000000  ...   -3.600000   -0.100000
50%    3.202217e+06        25.000000  ...   -1.300000    0.100000
75%    1.000000e+07        47.000000  ...    0.500000    0.900000
max    3.468255e+07        62.000000  ...   54.400000    8.600000

[8 rows x 25 columns]

Interesting columns: salary
max and min vary by 30mil, 
average salery 6.6 mil, with median 3.2 mil

In [ ]:
# describe objects
df.describe(include=['O'])

Player NBA_Country   Tm
count          485         485  485
unique         483          44   31
top     Kay Felder         USA  TOT
freq             3         374   55

This shows us, player Key Fedler is put into the dataset 3 times as the 'unique' in player should equal the 'count'.
TOT in 'Tm' is a category for players that player for multiple teams in one season (eg. got traded during the season) this is reason why we see 31 unique values in 'Tm' since there are only 30 teams in nba, but TOT category is also included in "Tm"

This also shows that 374 players (out of 485) are from USA. 

In [ ]:
df["Tm"].value_counts()

TOT    55
DAL    18
MEM    17
UTA    17
GSW    16
ATL    16
IND    16
PHI    16
MIL    15
BOS    15
ORL    15
MIA    15
SAS    15
LAL    15
DET    14
NOP    14
PHO    14
CHI    14
OKC    14
CHO    14
POR    14
DEN    14
NYK    14
MIN    13
TOR    13
WAS    13
HOU    13
LAC    12
SAC    12
BRK    12
CLE    10
Name: Tm, dtype: int64

In [ ]:
# drop Key Fedler
df.drop_duplicates(subset="Player",keep=False, inplace=True)

In [ ]:
df.describe(include=['O'])

Player NBA_Country   Tm
count           482         482  482
unique          482          44   31
top     Jeff Withey         USA  TOT
freq              1         371   54

Now we can see no player is duplicated in df

In [ ]:
# Binning salaries into high/med/low
bins=np.linspace(min(df["Salary"]),max(df['Salary']),4)
group_name=['Low','Medium','High']
df['Salary_bin']=pd.cut(df['Salary'],bins, labels=group_name,include_lowest=True)

In [ ]:
df.head()

Player    Salary NBA_Country  ...  VORP  Salary-bin Salary_bin
0        Zhou Qi    815615       China  ...  -0.2         Low        Low
1  Zaza Pachulia   3477600     Georgia  ...   0.7         Low        Low
2  Zach Randolph  12307692         USA  ...   0.0      Medium     Medium
3    Zach LaVine   3202217         USA  ...  -0.1         Low        Low
4   Zach Collins   3057240         USA  ...  -0.2         Low        Low

[5 rows x 30 columns]

In [ ]:
# distribution of salaries
df['Salary_bin'].value_counts()

Low       378
Medium     81
High       23
Name: Salary_bin, dtype: int64

In [ ]:
missing_data=df.isnull().sum()

In [ ]:
for column in missing_data.columns.values.tolist():
  print(column)
  print(missing_data[column].value_counts())

AttributeError: ignored